gamemodes/terrortown/gamemode/shared/sh_player_ext.lua

```diff
@@ -272,9 +275,25 @@ end
 ---
 -- Returns the current @{ROLE}'s team
 -- @return[default=TEAM_NONE] string
+-- @note If the real role has the flag `alone`, TEAM_NONE will be returned.
+-- Use @{Player:GetRealTeam()} instead if this behavior is not intended.
 -- @realm shared
 function plymeta:GetTeam()
-	return self.roleteam or TEAM_NONE
+	local tm = self.roleteam
+
+	if tm ~= nil and not TEAMS[tm].alone then
+		return tm
+	end
+
+	return TEAM_NONE
+end
+
+---
+-- Returns the real @{ROLE}'s team
+-- @return ?string
```

Comment:
does the `?` have a special meaning i'm not aware of?

Reply:
`nil|string`